activerecord/lib/active_record/autosave_association.rb

```diff
@@ -366,6 +371,12 @@ def normalize_reflection_attribute(indexed_attribute, reflection, index, attribu
         end
       end
 
+      def around_save_mark_as_saving_record
+        @saving = true
```

Comment:
Perhaps it would be a good practise to name this variable as verbose as possible and perhaps prefix it with an underscore so there is less risk of a collision with instance variables defined by the application. Perhaps something like `@_marked_as_saving`

Reply:
agreed, thanks for suggestion